src/qt/window.cpp

```diff
@@ -1582,6 +1585,39 @@ QWidget *wxWindowQt::GetHandle() const
     return m_qtWindow;
 }
 
+void wxWindowQt::SetMinSize(const wxSize& minSize)
+{
+    wxWindowBase::SetMinSize(minSize);
+    const int width = minSize.GetWidth();
+    const int height = minSize.GetHeight();
+    QWidget* widget = GetHandle();
+    if ( widget != NULL )
+       widget->setMinimumSize(QSize(width == -1 ? 0 : width, height == -1 ? 0 : height));
+}
+
+void wxWindowQt::SetMaxSize(const wxSize& maxSize)
+{
+    wxWindowBase::SetMaxSize(maxSize);
+    const int width = maxSize.GetWidth();
+    const int height = maxSize.GetHeight();
+    QWidget* widget = GetHandle();
+    if ( widget != NULL )
+        widget->setMaximumSize(QSize(width == -1 ? QWIDGETSIZE_MAX : width, height == -1 ? QWIDGETSIZE_MAX : height));
+}
+
+void wxWindowQt::DoSetSizeHints(int minW, int minH,
+    int maxW, int maxH,
+    int incW, int incH)
+{
+    wxWindowBase::DoSetSizeHints(minW, minH, maxW, maxH, incW, incH);
+    QWidget *widget = GetHandle();
+    if ( widget != NULL )
+    {
+        widget->setMinimumSize(QSize(std::max(minW, 0), std::max(minH, 0)));
```

Comment:
First, minor, point is that we should avoid calling this entirely if `minW == -1 && minH == -1`.

Second, more important one, is that this is almost certainly not the right function to call anyhow. We need to use some TLW-specific function which prevents the user from resizing the window to be smaller. I'm not sure if `minimumSizeHint` is the right Qt property to use, its name is similar, but it doesn't seem to fit neither.